### PR TITLE
convert color mixing methods to Swift

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Convert `UIColor+Mixing` helper methods to Swift.

### dependency
- [x] wait for https://github.com/wireapp/wire-ios/pull/3768 merge and clean up the code.
- [x] refactor `wr_color` to a `init` method

### TODO:
- [ ] move color helper methods to utilities framwork

### CODE REVIEW

File: Wire-iOS Tests/Color/ColorComponentsTests.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        let redColor = UIColor.red
+        
+        ///WHEN
+        let sut = UIColor.Components(color: redColor)
+        
+        ///THEN
+        XCTAssertEqual(sut, redColor.components)
+    }
+
+    func testThatColorCanNotBeCompareDirectly() {
```
Comment (johnxnguyen) [inline Wire-iOS Tests/Color/ColorComponentsTests.swift]: Why should we not be able to compare colors directly?

File: Wire-iOS/Sources/Helpers/UIColor+Equaltable.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    /// Create a color with a tuple rgba. The range of each component is 0 to 255 and alpha 1
+    ///
+    /// - Parameter rgba: tuple of color components
+    convenience init(rgb: (red: CGFloat, green: CGFloat, blue: CGFloat)) {
+        self.init(rgba: (red: rgb.red,
+                         green: rgb.green,
+                         blue: rgb.blue,
+                         alpha: 1))///TODO: test
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/Helpers/UIColor+Equaltable.swift]: Todo?

File: Wire-iOS/Sources/Helpers/UIColor+Mixing.swift [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        let component0 = components
+        let component1 = color.components
+
+        let red = mix(value0: component0.red, value1: component1.red, progress: progress)
+        let green = mix(value0: component0.green, value1: component1.green, progress: progress)
+        let blue = mix(value0: component0.blue, value1: component1.blue, progress: progress)
+        let alpha = mix(value0: component0.alpha, value1: component1.alpha, progress: progress)
+        
+        return UIColor(red: red, green: green, blue: blue, alpha: alpha)
+    }///TODO: test
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/Helpers/UIColor+Mixing.swift]: Todo?

File: Wire-iOS/Sources/Helpers/UIColor+Mixing.swift [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        let component0 = components
+        let component1 = color.components
+        let alpha0 = component0.alpha
+        
+        let red = mix(value0: component1.red, value1: component0.red, progress: alpha0)
+        let green = mix(value0: component1.green, value1: component0.green, progress: alpha0)
+        let blue = mix(value0: component1.blue, value1: component0.blue, progress: alpha0)
+        
+        return UIColor(red: red, green: green, blue: blue, alpha: 1)
+    }///TODO: test
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/Helpers/UIColor+Mixing.swift]: Todo?

File: Wire-iOS Tests/Color/ColorComponentsTests.swift (hunk 1 above)
Comment (billypchan) [reply Wire-iOS Tests/Color/ColorComponentsTests.swift]: `UIColor` does not conform to `Equatable` and we need to create our own compare method.

**Comment by billypchan** [comment]
@johnxnguyen tests added and I will move the helper methods to utilities after approval.

**Pull request merged**